### PR TITLE
feat: save storage secrets 

### PR DESCRIPTION
**Pull request by olevski**
Migrating
https://github.com/SwissDataScienceCenter/renku-notebooks/pull/1927

Depends on #566 and #565 

Also will keep this as draft and resolve/merge only when the new data connectors code and secrets that @leafty is working on is merged.

/deploy renku-ui=main renku-gateway=feat-shutdown-notebooks renku-notebooks=feat-use-different-user-labels-in-watcher

### CODE REVIEW

**Comment by Panaetius** [comment]
would this be a good time to switch to direct calls in code instead of calling the data svc REST api from within data svc?

**Comment by olevski** [comment]
> would this be a good time to switch to direct calls in code instead of calling the data svc REST api from within data svc?

I would rather do this in a separate pr. I really dont want to bloat these PRs with anything else other than moving the code over form the notebook service. Then if you want to dig out what or why we did something it will be easier.

**Comment by olevski** [comment]
Here is a follow up PR for this: https://github.com/SwissDataScienceCenter/renku-data-services/issues/586

**Comment by olevski** [comment]
This is ready for review. The stuff in the pr description is needed only to be able to test and direct requests to the data service instead of the notebook service.

Also I removed http calls to the data service for cloud storage. Which were there prior to this from the code that got moved over from notebooks.

File: components/renku_data_services/notebooks/api/schemas/cloud_storage.py [hunk 1]
```diff
@@ -57,54 +56,55 @@ def __init__(
...
         else:
             source_path = data["source_path"]
             target_path = data["target_path"]
             configuration = data["configuration"]
             readonly = data.get("readonly", True)
-        mount_folder = str(work_dir / target_path)
 
-        await config.storage_validator.validate_storage_configuration(configuration, source_path)
-        return cls(source_path, configuration, readonly, mount_folder, name, config)
+        # NOTE: This is used only in Renku v1, there we cannot save secrets for storage
```
Comment (leafty) [inline components/renku_data_services/notebooks/api/schemas/cloud_storage.py:95]:
```suggestion
        # NOTE: This is used only in Renku v1, there we do not save secrets for storage
```

We could save them but we don't.

**Review by leafty — APPROVED** [review]
LGTM, seems to be working for renku 1.0 sessions.

**Pull request merged**